src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenerationContext.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Uno.Extensions.Generators;
+using Uno.Extensions.Navigation.UI.Controls;
+
+namespace Uno.Extensions.Navigation.Generators;
+
+internal record ForceBindingsUpdateGenerationContext(
+	GeneratorExecutionContext Context,
+
+	// Core types
+	[ContextType("Uno.Extensions.Navigation.IForceBindingsUpdate")] INamedTypeSymbol ForceBindingsUpdateInterface,
+
+	// Attributes
+	[ContextType(typeof(ForceUpdateAttribute))] INamedTypeSymbol UpdateAttribute,
+
+	// General stuff types
+	[ContextType("Microsoft.UI.Xaml.Controls.Page")] INamedTypeSymbol Page)
+{
+	public bool IsGenerationNotDisable(ISymbol symbol)
+		=> IsGenerationEnabled(symbol) ?? true;
+
+	public bool? IsGenerationEnabled(ISymbol symbol)
+		=> symbol.FindAttributeValue<bool>(UpdateAttribute, nameof(ForceUpdateAttribute.IsEnabled), 0) is { isDefined: true } attribute
+			? attribute.value ?? true
+			: null;
+
+	private IImmutableSet<string>? _XBindFiles;
+	private static Regex ClassRegEx = new Regex("x:Class=\"([\\w.]+)\"");
+	private static Regex xBindRegEx = new Regex("{x:Bind ");
+
+	private IImmutableSet<string> XBindFiles
+	{
+		get
+		{
+			if (_XBindFiles is null)
+			{
+				var files = new List<string>();
+				foreach (var file in Context.AdditionalFiles)
+				{
+					var txt = file.GetText()?.ToString();
```

Comment:
This can be very expensive (noticeably), especially for large XAML files (e.g, when [XamlMerge](https://platform.uno/docs/articles/external/uno.xamlmerge.task/doc/using-xamlmerge.html) is used.

Would you be able to test the package produced in PR artifacts in a reasonable-sized project and evaluate the performance?